torch/_refs/_conversions.py

```diff
@@ -61,3 +70,48 @@ def fn(
 long = _make_conversion_method("long", torch.long)
 
 short = _make_conversion_method("short", torch.short)
+
+
+@register_decomposition(torch.ops.aten.complex)
+# Note: complex has type promotion tests disabled due to different semantics.
+# exact_dtype is for compat with complex_check_dtype from core.
+@out_wrapper(exact_dtype=True)
+def complex(real: TensorLikeType, imag: TensorLikeType) -> TensorLikeType:
+    def _allowed_dtypes():
+        return {torch.float32, torch.float64, torch.float16}
+
+    check(
+        real.dtype in _allowed_dtypes() and imag.dtype in _allowed_dtypes(),
+        lambda: (
+            f"Expected both inputs to be Half, Float or Double tensors but got "
+            f"{real.dtype} and {imag.dtype}"
+        ),
+    )
+    check(
```

Comment:
if you put this check first, in the second check you simply need to do `real.dtype in (torch.float32, torch.float64, torch.float16)`. Also, for small sets it's not necessary to use a set, simply use a tuple.

Reply:
I don't want to re-order the checks because they are in this order in core. Otherwise, I'd need to add special-casing in error inputs, which is not worth it for this IMO. In core, it's like this for a reason -- one of them is in `out`. So not going to do it.